lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1448,6 +1489,42 @@ func (w *WalletKit) ListAccounts(ctx context.Context,
 	return &ListAccountsResponse{Accounts: rpcAccounts}, nil
 }
 
+// ListAddresses retrieves all the addresses along with their balance. An
+// account name filter can be provided to filter through all of the
+// wallet accounts and return the addresses of only those matching.
+func (w *WalletKit) ListAddresses(ctx context.Context,
+	req *ListAddressesRequest) (*ListAddressesResponse, error) {
+
+	addressLists, err := w.cfg.Wallet.ListAddresses(req.AccountName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a slice of accounts from addressLists map.
+	accounts := make([]*waddrmgr.AccountProperties, 0, len(addressLists))
+	for account := range addressLists {
+		accounts = append(accounts, account)
+	}
+
+	// Sort the accounts by account number.
+	sort.Slice(accounts, func(i, j int) bool {
+		return accounts[i].AccountNumber < accounts[j].AccountNumber
```

Comment:
I think we need to sort by derivation path to make this a bit more stable. Because the account numbers can repeat within the different key scopes.